File: Crio/crio_programming_interview_problems-master/BinaryTreeZigzagLevelOrderTraversal/BinaryTreeZigzagLevelOrderTraversal.cpp
#include <bits/stdc++.h>
#include "../crio/cpp/ds/TreeNode/TreeNode.hpp"
#include "../crio/cpp/io/PrintMatrix.hpp"
#include "Solution.cpp"
using namespace std;

// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS

int main()
{
	int tests;
	cin >> tests;
	while (tests--)
	{
		TreeNode* root  = TreeNode().readTreeReturnRoot();
		vector<vector<int>> result =
		                     BinaryTreeZigzagLevelOrderTraversal().zigzagLevelOrder(root);
		PrintMatrix<int>().TwoDMatrix(result);


	}
	return 0;
}

File: Crio/StackAndQueue/5_BinaryTreeZigzagLevelOrderTraversal/BinaryTreeZigzagLevelOrderTraversal.py
from crio.io import PrintMatrix
from crio.ds.Tree.TreeNode import TreeNode, readTreeReturnRoot
from Solution import zigzagLevelOrder 

# CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
# CRIO_SOLUTION_END_MODULE_L1_PROBLEMS


def main():
    tests = int(input())
    for _ in range(tests):
        root  = readTreeReturnRoot()
        result = zigzagLevelOrder(root)
        PrintMatrix.TwoDMatrix(result)

if __name__ == '__main__':
    import sys
    sys.setrecursionlimit(30 * 1000)
    main()
